devtools/dbt_helper.py

```diff
@@ -98,6 +109,21 @@ class DbtSchema(BaseModel):
     sources: list[DbtSource]
     models: list[DbtTable] | None = None
 
+    def add_source_tests(
+        self, source_tests: list, model_name: str | None = None
+    ) -> "DbtSchema":
+        """Add data tests to source in dbt config."""
+        if model_name is None:
+            schema = self.model_copy(
+                update={"sources": [self.sources[0].add_source_tests(source_tests)]}
+            )
+        else:
+            models = {model.name: model for model in self.models}
+            models[model_name] = models[model_name].add_source_tests(source_tests)
+            schema = self.model_copy(update={"models": list(models.values())})
+
+        return schema
```

Comment:
Can you explain what these `add_source_tests` are doing?

Reply:
`dbt_tests` is going to stick around as part of the schema drift checker in #4145, so making/keeping it nice is worth it imo